src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/Designer/LaunchProfileEditorServiceFactory.cs

```diff
@@ -0,0 +1,93 @@
+{
+    [Export(typeof(IPackageService))]
+    internal sealed class LaunchProfileEditorServiceFactory : IPackageService, IDisposable
+    {
+        private const string LaunchProfileEditorServiceName = "Microsoft.VisualStudio.ProjectSystem.Managed.LaunchProfileEditorService";
+        private const string ProjectGuidActivationArgumentName = "ProjectGuid";
+
+        internal static readonly ServiceJsonRpcDescriptor LaunchProfileEditorServiceDescriptorV1 = new(
+            new ServiceMoniker(LaunchProfileEditorServiceName, new Version(0, 1)),
+            ServiceJsonRpcDescriptor.Formatters.UTF8,
+            ServiceJsonRpcDescriptor.MessageDelimiters.HttpLikeHeaders);
+
+        private readonly IProjectServiceAccessor _projectServiceAccessor;
+        private IDisposable? _profferDisposable;
+        private ILaunchProfileEditorService? _launchProfileEditorService;
+
+        [ImportingConstructor]
+        public LaunchProfileEditorServiceFactory(IProjectServiceAccessor projectServiceAccessor)
+        {
+            _projectServiceAccessor = projectServiceAccessor;
+        }
+
+        public async Task InitializeAsync(IAsyncServiceProvider asyncServiceProvider)
+        {
+            IBrokeredServiceContainer brokeredServiceContainer = await asyncServiceProvider.GetServiceAsync<SVsBrokeredServiceContainer, IBrokeredServiceContainer>();
+
+            _profferDisposable = brokeredServiceContainer.Proffer(LaunchProfileEditorServiceDescriptorV1, CreateInstanceAsync);
+
+            async ValueTask<object?> CreateInstanceAsync(
+                ServiceMoniker moniker,
+                ServiceActivationOptions options,
+                IServiceBroker serviceBroker,
+                AuthorizationServiceClient authorizationServiceClient,
+                CancellationToken cancellationToken)
+            {
+                await authorizationServiceClient.AuthorizeOrThrowAsync(WellKnownProtectedOperations.CreateClientIsOwner(), cancellationToken);
+
+                if (options.ActivationArguments is null
+                    || !options.ActivationArguments.TryGetValue("ProjectGuid", out string projectGuidString))
+                {
+                    throw new InvalidOperationException($"Missing required activation argument \"{ProjectGuidActivationArgumentName}\".");
+                }
+
+                if (!Guid.TryParse(projectGuidString, out Guid projectGuid))
+                {
+                    throw new InvalidOperationException($"Unable to parse activation argument \"{ProjectGuidActivationArgumentName}\": \"{projectGuidString}\".");
+                }
+
+                if (options.ClientRpcTarget is not ILaunchProfileEditorClientSession clientSession)
+                {
+                    throw new InvalidOperationException($"Missing client RPC target \"{nameof(ILaunchProfileEditorClientSession)}\".");
+                }
+
+                if (_launchProfileEditorService is null)
```

Comment:
Here we obtain/create the MEF-exported `ILaunchProfileEditorService` so we can delegate to that.